presto-main/src/main/java/com/facebook/presto/operator/scalar/BitwiseFunctions.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.operator.Description;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.type.SqlType;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+public final class BitwiseFunctions
+{
+    private BitwiseFunctions() {}
+
+    @Description("count number of set bits in 2's complement representation")
+    @ScalarFunction
+    @SqlType(StandardTypes.BIGINT)
+    public static long bitCount(@SqlType(StandardTypes.BIGINT) long num, @SqlType(StandardTypes.BIGINT) long bits)
+    {
+        if (bits == 64) {
+            return Long.bitCount(num);
+        }
+        if (bits <= 1) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "2nd argument of bitcount must be at least 2");
+        }
+        long mask = ((1 << (bits - 1)) - 1) | 0x8000_0000_0000_0000L; // set the sign bit and the least (bits - 1) bits
```

Comment:
This should fail if the value is needs more "bits" to be represented than we are using.